### DISCUSSION TITLE
why does netbox-dns allow duplicate records to be created?

### DISCUSSION THREAD

**Discussion by john2893** [auroraresearchlab/netbox-dns#306] [Q&A] [answered]
I created so many if and else conditions to check and left the A records check as it is because A records when created again said already exists.
Now A records gets created even if it exists.
I created a check of TXT, NS, MX and SRV. I'm pretty sure this issue would have been solved long back.
Anyone help me with some suggestions please.

**Comment by peteeckel** [comment, marked as answer]
Because there is nothing in any RFC that prohibits duplicate A (and many other, though not all) records from being created.

The successor of `netbox-dns`, `netbox-plugin-dns`, has an optional setting to disallow duplicate records. `netbox-dns` does not, and since the project is without maintainers it won't be added, ever.

Also see [here](https://github.com/peteeckel/netbox-plugin-dns/issues/38) and [here](https://github.com/peteeckel/netbox-plugin-dns/pull/44).
